services/payments/datastore.go

```diff
@@ -26,32 +26,33 @@ import (
 	"github.com/google/uuid"
 )
 
-// Transaction - the main type explaining a transaction, type used for qldb via ion.
-type Transaction struct {
-	ID                  *uuid.UUID       `json:"idempotencyKey,omitempty" valid:"required"`
-	Amount              *ion.Decimal     `json:"amount" valid:"required"`
-	To                  string           `json:"to,omitempty" valid:"required"`
-	From                string           `json:"from,omitempty" valid:"required"`
-	Custodian           string           `json:"custodian,omitempty" valid:"in(uphold|gemini|bitflyer)"`
-	State               TransactionState `json:"state" valid:"required"`
-	DocumentID          string           `json:"documentId,omitempty"`
-	AttestationDocument string           `json:"attestation,omitempty"`
-	PayoutID            string           `json:"payoutId" valid:"required"`
-	Signature           string           `json:"signature" valid:"required"` // KMS signature only enclave can sign
-	Authorizations      []Authorization  `json:"authorizations"`
-	PublicKey           string           `json:"publicKey" valid:"required"` // KMS signature only enclave can sign
-	Currency            string           `json:"currency"`
-	DryRun              *string          `json:"dryRun"` // determines dry-run
-}
-
-// Authorization represents a single authorization from a payment authorizer indicating that
+type AuthenticatedPaymentState struct {
+	PaymentDetails
+	Status         PaymentStatus
+	Authorizations []PaymentAuthorization `json:"authorizations"`
+	DryRun         *string                `json:"dryRun"` // determines dry-run
+	LastError      *PaymentError
+	documentID     string
+}
+
+type PaymentDetails struct {
+	Amount    *ion.Decimal `json:"amount" valid:"required"`
+	To        string       `json:"to,omitempty" valid:"required"`
+	From      string       `json:"from,omitempty" valid:"required"`
+	Custodian string       `json:"custodian,omitempty" valid:"in(uphold|gemini|bitflyer)"`
+	PayoutID  string       `json:"payoutId" valid:"required"`
+	Currency  string       `json:"currency"`
```

Comment:
I don't see it in this PR but if we are setting `govalidator.SetFieldsRequiredByDefault(true)` which we do in the grant cmd setup router function then this will as validation is required